ikasaneip/builder/jar/src/test/java/org/ikasan/builder/FlowBuilderTest.java

```diff
@@ -1530,4 +1530,97 @@ public void test_successful_withMonitor_implied_build()
+     */
+    @Test
+    public void test_successful_consumer_implied_and_explicit_builds()
+    {
+        BuilderFactory builderFactory = ikasanApplication.getBuilderFactory();
+
+        // implied
+        builderFactory.getFlowBuilder("moduleName", "flowName")
+                .consumer("consumerName", builderFactory.getComponentBuilder().scheduledConsumer())
+                .producer("producerName", producer)
+                .build();
+
+        // explicit
+        builderFactory.getFlowBuilder("moduleName", "flowName")
+                .consumer("consumerName", builderFactory.getComponentBuilder().scheduledConsumer().build())
+                .producer("producerName", producer)
+                .build();
+
+        // implied
+        builderFactory.getFlowBuilder("moduleName", "flowName")
+                .consumer("consumerName", builderFactory.getComponentBuilder().fileConsumer())
+                .producer("producerName", producer)
+                .build();
+
+        // explicit
+        builderFactory.getFlowBuilder("moduleName", "flowName")
+                .consumer("consumerName", builderFactory.getComponentBuilder().fileConsumer().build())
+                .producer("producerName", producer)
+                .build();
+
+        // implied
+        builderFactory.getFlowBuilder("moduleName", "flowName")
+                .consumer("consumerName", builderFactory.getComponentBuilder().dbConsumer())
+                .producer("producerName", producer)
+                .build();
+
+        // explicit
+        builderFactory.getFlowBuilder("moduleName", "flowName")
+                .consumer("consumerName", builderFactory.getComponentBuilder().dbConsumer().build())
+                .producer("producerName", producer)
+                .build();
+
+//        // implied
+//        builderFactory.getFlowBuilder("moduleName", "flowName")
+//                .consumer("consumerName", builderFactory.getComponentBuilder().ftpConsumer())
+//                .producer("producerName", producer)
+//                .build();
+//
+//        // explicit
+//        builderFactory.getFlowBuilder("moduleName", "flowName")
+//                .consumer("consumerName", builderFactory.getComponentBuilder().ftpConsumer().build())
+//                .producer("producerName", producer)
+//                .build();
+
+        // implied
+        builderFactory.getFlowBuilder("moduleName", "flowName")
+                .consumer("consumerName", builderFactory.getComponentBuilder().jmsConsumer())
+                .producer("producerName", producer)
+                .build();
+
+        // explicit
+        builderFactory.getFlowBuilder("moduleName", "flowName")
+                .consumer("consumerName", builderFactory.getComponentBuilder().jmsConsumer().build())
+                .producer("producerName", producer)
+                .build();
+
+//        // implied
+//        builderFactory.getFlowBuilder("moduleName", "flowName")
```

Comment:
How comes this is left in commented out?

Reply:
Yep - my bad - this test is failing due to unrelated issues, but will see if I can fix it